### ISSUE TITLE
Run shell/tab as elevated different user

### ISSUE THREAD

**Issue by hahndorf** [cbucher/console#167]
My normal user is not an administrator, but I often need to run a shell as an elevated administrator.

Currently you can specify to run the current user elevated, but that is not helpful for me, I need to be able to run a tab/shell as a different user in elevated mode.

**Comment by cbucher** [comment]
> I need to be able to run a tab/shell as a different user in elevated mode.

If you choose "run as administrator" and your user is not an administrator then UAC prompts you to enter an administrator's credentials. This is how UAC works.

**Comment by hahndorf** [comment]
If theory yes, that's how UAC works, however if a choose "Run as current user" and "Administrator" but the current user is not in the administrators groups, I get no popup.
If I use "Run as user:" and choose and admin user, I get a popup asking for the user's password and the shell opens, but it is running with an Integrity level of medium, not high or elevated

**Comment by cbucher** [comment]
Behavior should be the same as when you launch an executable from explorer (right click on exe file, run as administrator).
What is your OS. What is your UAC level.

**Comment by hahndorf** [comment]
OS is Windows 8.1 with update (64bit), UAC level is the top most "Always notify me when..."
In Explorer when using "run as administrator", I can pick an admin user, enter the password and that process then runs elevated as expected.

**Comment by cbucher** [comment]
@hahndorf try experimental version 1.13.0.15003 or more (https://github.com/cbucher/console/wiki/Downloads)

**Comment by hahndorf** [comment]
This seems to work fine now, thanks Christophe.

**Comment by hahndorf** [comment]
So this works fine on a standalone machine where you can pick an admin user from a list. However when inside a domain, you have to type username and password. It would be nice if I could save the username with the tab and have a checkbox to run elevated. So Rather than having a radio button for administrator have another checkbox like "Net only" saying "as Admininistrator" or "elevated". This could then also be applied when run as "current user". In case that user is in the administrators group it would run elevated. I usually have three shells open, one run as standard user, one as local admin and one as domain admin. Having three tabs pre-configured for those users would be great.

**Comment by cbucher** [comment]
Create a process with elevated rights and create a process with different user credentials are two distinct actions (with distinct API !).

Your request can only be coded like this:
ConsoleZ A -> ConsoleZ B -> ConsoleZ C -> Shell

- ConsoleZ A is ConsoleZ process running with your credentials.
- ConsoleZ B is ConsoleZ process launched by ConsoleZ A with different user credentials. (**you are prompted for password here**)
- ConsoleZ C  is ConsoleZ process launched by ConsoleZ B with elevated privileges.  (**you are prompted for consent here**)
- Shell is launched by ConsoleZ C with current credentials (so with different user and  elevated privileges)

Personally if I have to choose between
- be prompted by UAC one time, entre user and password
- or be prompted by UAC, enter password, be prompted by UAC for consent

I choose the first (=default usage of UAC).

**Comment by cbucher** [comment]
@hahndorf I await your opinion.

**Comment by hahndorf** [comment]
@cbucher - I guess I understand, you can not impersonate and elevate at the same time. So think it's best to use a single prompt. This is how you implemented it in 1.13.0.15003

If in that version I specify a user to run as (not elevated) I get prompted for the password (username is pre-filled) but the resulting shell is still running as my original user. I'm sure used to work in previous versions.

**Comment by cbucher** [comment]
>I guess I understand, you can not impersonate and elevate at the same time. So think it's best to use a single prompt. This is how you implemented it in 1.13.0.15003

Yes it was my first opinion.
But the prompt for the password is not in UAC.
So we could save password in the vault
http://www.tech-recipes.com/rx/31797/windows-8-manage-saved-passwords-for-applications/
In this case you shall enter the password only the first time or when it changes.
I will do some tests.

>If in that version I specify a user to run as (not elevated) I get prompted for the password (username is pre-filled) but the resulting shell is still running as my original user. I'm sure used to work in previous versions.

@hahndorf You checked "Net only" ?

**Comment by hahndorf** [comment]
No, here's an example: 
console shell="%SYSTEMROOT%\System32\WindowsPowerShell\v1.0\Powershell.exe" init_dir="%bindir%" priority="2" run_as_user="1" user="peter" net_only="0" run_as_admin="0"

**Comment by cbucher** [comment]
Sorry but it works fine for me ...
(verify by typing whoami in your powershell session)

![image](https://cloud.githubusercontent.com/assets/363759/5690739/8a609f2e-98a0-11e4-8860-152cff8acce6.png)

![image](https://cloud.githubusercontent.com/assets/363759/5690731/e399b572-989f-11e4-85ee-66fa353dce9f.png)

![image](https://cloud.githubusercontent.com/assets/363759/5690760/c92e67f8-98a1-11e4-8f04-3e3008d29e6f.png)

**Comment by hahndorf** [comment]
I tried on another Windows 8.1, a fresh Windows 10 and a newly set up Windows 7. In all cases the new shell runs under the original user. The main difference is the prompt, in all cases I get:
![usernamepassword](https://cloud.githubusercontent.com/assets/778685/5690970/8c7efe40-98af-11e4-86b8-6266ba897042.png)
which is different from the one you get. Whether the base user is an administrator or an elevated administrator doesn't make a difference. Same for the runas user. 
I started without a console.xml to have a fresh start.
